src/main/scala/io/qbeast/spark/utils/QbeastUtils.scala

```diff
@@ -72,10 +59,76 @@ object QbeastUtils extends Logging {
       .as[String]
       .collect()
 
-    log.info(s"Histogram for column $columnName: $histogram")
-    histogram
-      .map(string => s"'$string'")
-      .mkString("[", ",", "]")
+    log.info(s"String Quantiles for column $columnName: $quantiles")
+    quantiles.map(string => s"'$string'")
+
+  }
+
+  private def computeQuantilesForNumericColumn(
+      df: DataFrame,
+      columnName: String,
+      numberOfQuantiles: Int,
+      relativeError: Double): Array[Double] = {
+    val probabilities = (1 to numberOfQuantiles).map(_ / numberOfQuantiles.toDouble).toArray
+    val reducedDF = df.select(columnName)
+    val approxQuantile = reducedDF.stat.approxQuantile(columnName, probabilities, relativeError)
+    log.info(s"Numeric Quantiles for column $columnName: ${approxQuantile.mkString(",")}")
+    approxQuantile
+  }
+
+  /**
+   * Compute the quantiles for a given column
+   *
+   * Since computing the quantiles can be expensive, this method is used outside the indexing
+   * process.
+   *
+   * It outputs the quantiles of the column as format [bin1, bin2, bin3, ...] Number of bins by
```

Comment:
The numbers returned are not bins themselves but rather the boundaries of the bins.